metals/src/main/scala/scala/meta/internal/metals/FileDecoderProvider.scala

```diff
@@ -406,6 +425,70 @@ final class FileDecoderProvider(
     }
   }
 
+  private def decodeCFRFromClassFile(
+      path: AbsolutePath
+  ): Future[DecoderResponse] = {
+    Future {
+      Try {
+        val out = new ByteArrayOutputStream()
+        val err = new ByteArrayOutputStream()
+        val exceptions = new ListBuffer[Exception]()
+        val psOut = new PrintStream(out)
+        val psErr = new PrintStream(err)
+        try {
+          val sink = new OutputSinkFactory() {
+
+            override def getSupportedSinks(
+                sinkType: OutputSinkFactory.SinkType,
+                sinkClasses: Collection[OutputSinkFactory.SinkClass]
+            ): ju.List[OutputSinkFactory.SinkClass] =
+              List(
+                OutputSinkFactory.SinkClass.EXCEPTION_MESSAGE,
+                OutputSinkFactory.SinkClass.STRING
+              ).asJava
+
+            override def getSink[T](
+                sinkType: OutputSinkFactory.SinkType,
+                sinkClass: OutputSinkFactory.SinkClass
+            ): OutputSinkFactory.Sink[T] =
+              sinkType match {
+                case OutputSinkFactory.SinkType.JAVA => psOut.print(_)
```

Comment:
Instead of having `ByteArrayOutputStream` and `PrintStream` there is an option just to append into list.
Smth like:
```scala
val out = List.newBuilder[String]
def appendToOut[A](a: A): Unit = 
  a match {
    case s: String => out += s
    case _ =>
  }

....
  sinkType match {
    case OutputSinkFactory.SinkType.JAVA  => appendToOut
  }
```

In this case there you can remove an additional `try/finally` block as there wouldn't be a need to close writers.